==22508== Memcheck, a memory error detector
==22508== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==22508== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==22508== Command: /home/marija/Desktop/2024_Analysis_12-Uno/valgrind/memcheck/../../12-UNO/UNO/build/run_tests
==22508== Parent PID: 22024
==22508== 
==22508== Conditional jump or move depends on uninitialised value(s)
==22508==    at 0x13D826: Catch::RunContext::handleExpr(Catch::AssertionInfo const&, Catch::ITransientExpression const&, Catch::AssertionReaction&) (catch.hpp:13049)
==22508==    by 0x12C627: Catch::AssertionHandler::handleExpr(Catch::ITransientExpression const&) (catch.hpp:8230)
==22508==    by 0x202AAB: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:72)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==  Uninitialised value was created by a heap allocation
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x20299D: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:69)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== Conditional jump or move depends on uninitialised value(s)
==22508==    at 0x629AA7D: std::ostreambuf_iterator<char, std::char_traits<char> > std::num_put<char, std::ostreambuf_iterator<char, std::char_traits<char> > >::_M_insert_int<long>(std::ostreambuf_iterator<char, std::char_traits<char> >, std::ios_base&, char, long) const (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.33)
==22508==    by 0x62AB3C1: std::ostream& std::ostream::_M_insert<long>(long) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.33)
==22508==    by 0x1FA85B: Catch::ReusableStringStream& Catch::ReusableStringStream::operator<< <Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1451)
==22508==    by 0x1FA79D: std::enable_if<Catch::Detail::IsStreamInsertable<Card::CARD_TYPE>::value, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::type Catch::StringMaker<Card::CARD_TYPE, void>::convert<Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1623)
==22508==    by 0x1FA69E: std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > Catch::Detail::stringify<Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1645)
==22508==    by 0x1FA539: Catch::BinaryExpr<Card::CARD_TYPE const&, Card::CARD_TYPE const&>::streamReconstructedExpression(std::ostream&) const (catch.hpp:2231)
==22508==    by 0x12C3E9: Catch::(anonymous namespace)::operator<<(std::ostream&, Catch::ITransientExpression const&) (catch.hpp:8189)
==22508==    by 0x12C527: Catch::operator<<(std::ostream&, Catch::LazyExpression const&) (catch.hpp:8212)
==22508==    by 0x1713AA: Catch::ReusableStringStream& Catch::ReusableStringStream::operator<< <Catch::LazyExpression>(Catch::LazyExpression const&) (catch.hpp:1451)
==22508==    by 0x12CAFA: Catch::AssertionResultData::reconstructExpression[abi:cxx11]() const (catch.hpp:8301)
==22508==    by 0x12D02C: Catch::AssertionResult::getExpandedExpression[abi:cxx11]() const (catch.hpp:8367)
==22508==    by 0x12CF4B: Catch::AssertionResult::hasExpandedExpression() const (catch.hpp:8363)
==22508==  Uninitialised value was created by a heap allocation
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x20299D: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:69)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== Use of uninitialised value of size 8
==22508==    at 0x629A95B: ??? (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.33)
==22508==    by 0x629AAA7: std::ostreambuf_iterator<char, std::char_traits<char> > std::num_put<char, std::ostreambuf_iterator<char, std::char_traits<char> > >::_M_insert_int<long>(std::ostreambuf_iterator<char, std::char_traits<char> >, std::ios_base&, char, long) const (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.33)
==22508==    by 0x62AB3C1: std::ostream& std::ostream::_M_insert<long>(long) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.33)
==22508==    by 0x1FA85B: Catch::ReusableStringStream& Catch::ReusableStringStream::operator<< <Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1451)
==22508==    by 0x1FA79D: std::enable_if<Catch::Detail::IsStreamInsertable<Card::CARD_TYPE>::value, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::type Catch::StringMaker<Card::CARD_TYPE, void>::convert<Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1623)
==22508==    by 0x1FA69E: std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > Catch::Detail::stringify<Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1645)
==22508==    by 0x1FA539: Catch::BinaryExpr<Card::CARD_TYPE const&, Card::CARD_TYPE const&>::streamReconstructedExpression(std::ostream&) const (catch.hpp:2231)
==22508==    by 0x12C3E9: Catch::(anonymous namespace)::operator<<(std::ostream&, Catch::ITransientExpression const&) (catch.hpp:8189)
==22508==    by 0x12C527: Catch::operator<<(std::ostream&, Catch::LazyExpression const&) (catch.hpp:8212)
==22508==    by 0x1713AA: Catch::ReusableStringStream& Catch::ReusableStringStream::operator<< <Catch::LazyExpression>(Catch::LazyExpression const&) (catch.hpp:1451)
==22508==    by 0x12CAFA: Catch::AssertionResultData::reconstructExpression[abi:cxx11]() const (catch.hpp:8301)
==22508==    by 0x12D02C: Catch::AssertionResult::getExpandedExpression[abi:cxx11]() const (catch.hpp:8367)
==22508==  Uninitialised value was created by a heap allocation
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x20299D: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:69)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== Conditional jump or move depends on uninitialised value(s)
==22508==    at 0x629A96D: ??? (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.33)
==22508==    by 0x629AAA7: std::ostreambuf_iterator<char, std::char_traits<char> > std::num_put<char, std::ostreambuf_iterator<char, std::char_traits<char> > >::_M_insert_int<long>(std::ostreambuf_iterator<char, std::char_traits<char> >, std::ios_base&, char, long) const (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.33)
==22508==    by 0x62AB3C1: std::ostream& std::ostream::_M_insert<long>(long) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.33)
==22508==    by 0x1FA85B: Catch::ReusableStringStream& Catch::ReusableStringStream::operator<< <Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1451)
==22508==    by 0x1FA79D: std::enable_if<Catch::Detail::IsStreamInsertable<Card::CARD_TYPE>::value, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::type Catch::StringMaker<Card::CARD_TYPE, void>::convert<Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1623)
==22508==    by 0x1FA69E: std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > Catch::Detail::stringify<Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1645)
==22508==    by 0x1FA539: Catch::BinaryExpr<Card::CARD_TYPE const&, Card::CARD_TYPE const&>::streamReconstructedExpression(std::ostream&) const (catch.hpp:2231)
==22508==    by 0x12C3E9: Catch::(anonymous namespace)::operator<<(std::ostream&, Catch::ITransientExpression const&) (catch.hpp:8189)
==22508==    by 0x12C527: Catch::operator<<(std::ostream&, Catch::LazyExpression const&) (catch.hpp:8212)
==22508==    by 0x1713AA: Catch::ReusableStringStream& Catch::ReusableStringStream::operator<< <Catch::LazyExpression>(Catch::LazyExpression const&) (catch.hpp:1451)
==22508==    by 0x12CAFA: Catch::AssertionResultData::reconstructExpression[abi:cxx11]() const (catch.hpp:8301)
==22508==    by 0x12D02C: Catch::AssertionResult::getExpandedExpression[abi:cxx11]() const (catch.hpp:8367)
==22508==  Uninitialised value was created by a heap allocation
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x20299D: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:69)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== Conditional jump or move depends on uninitialised value(s)
==22508==    at 0x629AADC: std::ostreambuf_iterator<char, std::char_traits<char> > std::num_put<char, std::ostreambuf_iterator<char, std::char_traits<char> > >::_M_insert_int<long>(std::ostreambuf_iterator<char, std::char_traits<char> >, std::ios_base&, char, long) const (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.33)
==22508==    by 0x62AB3C1: std::ostream& std::ostream::_M_insert<long>(long) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.33)
==22508==    by 0x1FA85B: Catch::ReusableStringStream& Catch::ReusableStringStream::operator<< <Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1451)
==22508==    by 0x1FA79D: std::enable_if<Catch::Detail::IsStreamInsertable<Card::CARD_TYPE>::value, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::type Catch::StringMaker<Card::CARD_TYPE, void>::convert<Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1623)
==22508==    by 0x1FA69E: std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > Catch::Detail::stringify<Card::CARD_TYPE>(Card::CARD_TYPE const&) (catch.hpp:1645)
==22508==    by 0x1FA539: Catch::BinaryExpr<Card::CARD_TYPE const&, Card::CARD_TYPE const&>::streamReconstructedExpression(std::ostream&) const (catch.hpp:2231)
==22508==    by 0x12C3E9: Catch::(anonymous namespace)::operator<<(std::ostream&, Catch::ITransientExpression const&) (catch.hpp:8189)
==22508==    by 0x12C527: Catch::operator<<(std::ostream&, Catch::LazyExpression const&) (catch.hpp:8212)
==22508==    by 0x1713AA: Catch::ReusableStringStream& Catch::ReusableStringStream::operator<< <Catch::LazyExpression>(Catch::LazyExpression const&) (catch.hpp:1451)
==22508==    by 0x12CAFA: Catch::AssertionResultData::reconstructExpression[abi:cxx11]() const (catch.hpp:8301)
==22508==    by 0x12D02C: Catch::AssertionResult::getExpandedExpression[abi:cxx11]() const (catch.hpp:8367)
==22508==    by 0x12CF4B: Catch::AssertionResult::hasExpandedExpression() const (catch.hpp:8363)
==22508==  Uninitialised value was created by a heap allocation
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x20299D: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:69)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== Conditional jump or move depends on uninitialised value(s)
==22508==    at 0x13D826: Catch::RunContext::handleExpr(Catch::AssertionInfo const&, Catch::ITransientExpression const&, Catch::AssertionReaction&) (catch.hpp:13049)
==22508==    by 0x12C627: Catch::AssertionHandler::handleExpr(Catch::ITransientExpression const&) (catch.hpp:8230)
==22508==    by 0x203368: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:128)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==  Uninitialised value was created by a heap allocation
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x235E7E: Pile::lastCard() const (pile.cpp:31)
==22508==    by 0x20313B: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:123)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 
==22508== HEAP SUMMARY:
==22508==     in use at exit: 2,455,344 bytes in 24,411 blocks
==22508==   total heap usage: 326,127 allocs, 301,716 frees, 33,969,710 bytes allocated
==22508== 
==22508== 16 bytes in 1 blocks are still reachable in loss record 490 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5D23C4F: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5CA44ED: QThread::currentThread() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5B9019E: QCoreApplicationPrivate::QCoreApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x522B858: QGuiApplicationPrivate::QGuiApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x5237C2D: QGuiApplication::QGuiApplication(int&, char**, int) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x157332: main (main.cpp:7)
==22508== 
==22508== 20 bytes in 1 blocks are indirectly lost in loss record 1,619 of 7,517
==22508==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5CBB860: QArrayData::allocate(QArrayData**, long long, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5C89D2E: QString::QString(long long, Qt::Initialization) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5C9E3A9: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5C84613: QString::fromUtf8(QByteArrayView) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x1FDC5D: QString::operator=(char const*) (qstring.h:1035)
==22508==    by 0x1FC08C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:159)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508== 
==22508== 24 bytes in 1 blocks are indirectly lost in loss record 2,206 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF48: Deck::Deck() (deck.cpp:41)
==22508==    by 0x238ED3: Game::initDeck() (game.cpp:108)
==22508==    by 0x238B37: Game::setupBoard() (game.cpp:67)
==22508==    by 0x23894F: Game::Game(QObject*, GameParameters*) (game.cpp:27)
==22508==    by 0x1FC0B0: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:162)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508== 
==22508== 24 bytes in 1 blocks are indirectly lost in loss record 2,207 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF7C: Deck::Deck() (deck.cpp:42)
==22508==    by 0x238ED3: Game::initDeck() (game.cpp:108)
==22508==    by 0x238B37: Game::setupBoard() (game.cpp:67)
==22508==    by 0x23894F: Game::Game(QObject*, GameParameters*) (game.cpp:27)
==22508==    by 0x1FC0B0: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:162)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508== 
==22508== 24 bytes in 1 blocks are indirectly lost in loss record 2,208 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x2024F2: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:37)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are indirectly lost in loss record 2,209 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x2069BA: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:66)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are indirectly lost in loss record 2,210 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x2069E0: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:67)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are indirectly lost in loss record 2,211 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x206A01: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:68)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are indirectly lost in loss record 2,212 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x2089D9: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:323)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are indirectly lost in loss record 2,213 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x2089FF: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:324)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are indirectly lost in loss record 2,214 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x208A25: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:325)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are indirectly lost in loss record 2,215 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x20CC7B: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:19)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are definitely lost in loss record 2,216 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x20299D: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:69)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are definitely lost in loss record 2,217 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x235E7E: Pile::lastCard() const (pile.cpp:31)
==22508==    by 0x20313B: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:123)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 24 bytes in 1 blocks are definitely lost in loss record 2,218 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x2040DC: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:224)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are definitely lost in loss record 2,219 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x20629C: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:14)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are definitely lost in loss record 2,220 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x206994: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:65)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are definitely lost in loss record 2,221 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x209688: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:422)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 24 bytes in 1 blocks are definitely lost in loss record 2,222 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x20D487: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:80)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 32 bytes in 1 blocks are indirectly lost in loss record 3,520 of 7,517
==22508==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5CBB860: QArrayData::allocate(QArrayData**, long long, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5C89D2E: QString::QString(long long, Qt::Initialization) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5C9E3A9: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5C84613: QString::fromUtf8(QByteArrayView) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x1FDC5D: QString::operator=(char const*) (qstring.h:1035)
==22508==    by 0x1FC059: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:156)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508== 
==22508== 34 bytes in 1 blocks are indirectly lost in loss record 3,534 of 7,517
==22508==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5CBB860: QArrayData::allocate(QArrayData**, long long, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x20193E: QTypedArrayData<Card*>::allocate(long long, QArrayData::AllocationOption) (qarraydata.h:105)
==22508==    by 0x23314C: QArrayDataPointer<Card*>::allocateGrow(QArrayDataPointer<Card*> const&, long long, QArrayData::GrowthPosition) (qarraydatapointer.h:352)
==22508==    by 0x23280E: QArrayDataPointer<Card*>::reallocateAndGrow(QArrayData::GrowthPosition, long long, QArrayDataPointer<Card*>*) (qarraydatapointer.h:213)
==22508==    by 0x232B1A: QArrayDataPointer<Card*>::detachAndGrow(QArrayData::GrowthPosition, long long, Card* const**, QArrayDataPointer<Card*>*) (qarraydatapointer.h:194)
==22508==    by 0x232208: void QtPrivate::QPodArrayOps<Card*>::emplace<Card*&>(long long, Card*&) (qarraydataops.h:174)
==22508==    by 0x231707: Card*& QList<Card*>::emplaceBack<Card*&>(Card*&) (qlist.h:856)
==22508==    by 0x230E12: QList<Card*>::append(Card*) (qlist.h:433)
==22508==    by 0x230A08: QList<Card*>::push_back(Card*) (qlist.h:651)
==22508==    by 0x235E09: Pile::pushCard(Card*) (pile.cpp:17)
==22508==    by 0x20CCB5: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:20)
==22508== 
==22508== 48 bytes in 1 blocks are still reachable in loss record 3,906 of 7,517
==22508==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x6BD5B09: g_malloc (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==22508==    by 0x6BCE3A9: ??? (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==22508==    by 0x6BCFBC7: ??? (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==22508==    by 0x6BCFCF6: g_source_attach (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==22508==    by 0xBA7D69A: ??? (in /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.8000.0)
==22508==    by 0xBA7D847: ??? (in /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.8000.0)
==22508==    by 0x6C04531: ??? (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==22508==    by 0x6BFED91: ??? (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==22508==    by 0x6587AA3: start_thread (pthread_create.c:447)
==22508==    by 0x6614A33: clone (clone.S:100)
==22508== 
==22508== 56 bytes in 1 blocks are still reachable in loss record 4,471 of 7,517
==22508==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x6BD5B09: g_malloc (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==22508==    by 0xB046F96: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==22508==    by 0xB050B02: gtk_entry_set_icon_from_pixbuf (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==22508==    by 0xB04559A: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==22508==    by 0xBBBC6C3: g_object_run_dispose (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.8000.0)
==22508==    by 0xAEBCE7C: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==22508==    by 0xAEB31C4: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==22508==    by 0xAEBBF24: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==22508==    by 0x522B0C6: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x522B518: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x157380: main (main.cpp:7)
==22508== 
==22508== 56 bytes in 1 blocks are still reachable in loss record 4,473 of 7,517
==22508==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x6BD5B09: g_malloc (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==22508==    by 0xB046F96: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==22508==    by 0xB050B02: gtk_entry_set_icon_from_pixbuf (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==22508==    by 0xB0455B5: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==22508==    by 0xBBBC6C3: g_object_run_dispose (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.8000.0)
==22508==    by 0xAEBCE7C: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==22508==    by 0xAEB31C4: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==22508==    by 0xAEBBF24: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==22508==    by 0x522B0C6: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x522B518: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x157380: main (main.cpp:7)
==22508== 
==22508== 66 bytes in 1 blocks are still reachable in loss record 4,781 of 7,517
==22508==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5CBB860: QArrayData::allocate(QArrayData**, long long, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5BECDE1: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5D2BDD7: QThreadStorageData::get() const (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x54857A9: QFontCache::cleanup() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x522B03F: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x522B518: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x157380: main (main.cpp:7)
==22508== 
==22508== 66 bytes in 1 blocks are indirectly lost in loss record 4,783 of 7,517
==22508==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5CBB860: QArrayData::allocate(QArrayData**, long long, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x20193E: QTypedArrayData<Card*>::allocate(long long, QArrayData::AllocationOption) (qarraydata.h:105)
==22508==    by 0x23314C: QArrayDataPointer<Card*>::allocateGrow(QArrayDataPointer<Card*> const&, long long, QArrayData::GrowthPosition) (qarraydatapointer.h:352)
==22508==    by 0x23280E: QArrayDataPointer<Card*>::reallocateAndGrow(QArrayData::GrowthPosition, long long, QArrayDataPointer<Card*>*) (qarraydatapointer.h:213)
==22508==    by 0x232B1A: QArrayDataPointer<Card*>::detachAndGrow(QArrayData::GrowthPosition, long long, Card* const**, QArrayDataPointer<Card*>*) (qarraydatapointer.h:194)
==22508==    by 0x2370EF: QtPrivate::QCommonArrayOps<Card*>::growAppend(Card* const*, Card* const*) (qarraydataops.h:922)
==22508==    by 0x236F07: QList<Card*>::append(QList<Card*>::const_iterator, QList<Card*>::const_iterator) (qlist.h:804)
==22508==    by 0x236D6A: QList<Card*>::append(QList<Card*> const&) (qlist.h:445)
==22508==    by 0x2369EE: Player::setHand(QList<Card*> const&) (player.cpp:110)
==22508==    by 0x206AA9: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:72)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508== 
==22508== 66 bytes in 1 blocks are indirectly lost in loss record 4,784 of 7,517
==22508==    at 0x484DB80: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5CBB920: QArrayData::reallocateUnaligned(QArrayData*, void*, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x23358A: QTypedArrayData<Card*>::reallocateUnaligned(QTypedArrayData<Card*>*, Card**, long long, QArrayData::AllocationOption) (qarraydata.h:117)
==22508==    by 0x232F8C: QtPrivate::QPodArrayOps<Card*>::reallocate(long long, QArrayData::AllocationOption) (qarraydataops.h:259)
==22508==    by 0x2327F2: QArrayDataPointer<Card*>::reallocateAndGrow(QArrayData::GrowthPosition, long long, QArrayDataPointer<Card*>*) (qarraydatapointer.h:208)
==22508==    by 0x232B1A: QArrayDataPointer<Card*>::detachAndGrow(QArrayData::GrowthPosition, long long, Card* const**, QArrayDataPointer<Card*>*) (qarraydatapointer.h:194)
==22508==    by 0x232208: void QtPrivate::QPodArrayOps<Card*>::emplace<Card*&>(long long, Card*&) (qarraydataops.h:174)
==22508==    by 0x231707: Card*& QList<Card*>::emplaceBack<Card*&>(Card*&) (qlist.h:856)
==22508==    by 0x230E12: QList<Card*>::append(Card*) (qlist.h:433)
==22508==    by 0x230A08: QList<Card*>::push_back(Card*) (qlist.h:651)
==22508==    by 0x235E09: Pile::pushCard(Card*) (pile.cpp:17)
==22508==    by 0x208A8C: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:328)
==22508== 
==22508== 72 bytes in 3 blocks are indirectly lost in loss record 4,848 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEAB: Deck::Deck() (deck.cpp:33)
==22508==    by 0x238ED3: Game::initDeck() (game.cpp:108)
==22508==    by 0x238B37: Game::setupBoard() (game.cpp:67)
==22508==    by 0x23894F: Game::Game(QObject*, GameParameters*) (game.cpp:27)
==22508==    by 0x1FC0B0: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:162)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508== 
==22508== 76 (56 direct, 20 indirect) bytes in 1 blocks are definitely lost in loss record 4,866 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x1FBFF6: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:155)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 90 (32 direct, 58 indirect) bytes in 1 blocks are definitely lost in loss record 5,562 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x20CC34: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:17)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 96 bytes in 1 blocks are still reachable in loss record 5,594 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5D2C994: QWaitCondition::QWaitCondition() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5CA50A3: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5D23C5D: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5CA44ED: QThread::currentThread() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5B9019E: QCoreApplicationPrivate::QCoreApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x522B858: QGuiApplicationPrivate::QGuiApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x5237C2D: QGuiApplication::QGuiApplication(int&, char**, int) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x157332: main (main.cpp:7)
==22508== 
==22508== 96 bytes in 4 blocks are indirectly lost in loss record 6,412 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEE7: Deck::Deck() (deck.cpp:34)
==22508==    by 0x238ED3: Game::initDeck() (game.cpp:108)
==22508==    by 0x238B37: Game::setupBoard() (game.cpp:67)
==22508==    by 0x23894F: Game::Game(QObject*, GameParameters*) (game.cpp:27)
==22508==    by 0x1FC0B0: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:162)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508== 
==22508== 96 bytes in 4 blocks are definitely lost in loss record 6,426 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF48: Deck::Deck() (deck.cpp:41)
==22508==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 96 bytes in 4 blocks are definitely lost in loss record 6,427 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF7C: Deck::Deck() (deck.cpp:42)
==22508==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 96 bytes in 4 blocks are definitely lost in loss record 6,428 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF48: Deck::Deck() (deck.cpp:41)
==22508==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 96 bytes in 4 blocks are definitely lost in loss record 6,429 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF7C: Deck::Deck() (deck.cpp:42)
==22508==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 96 bytes in 4 blocks are definitely lost in loss record 6,430 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF48: Deck::Deck() (deck.cpp:41)
==22508==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 96 bytes in 4 blocks are definitely lost in loss record 6,431 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF7C: Deck::Deck() (deck.cpp:42)
==22508==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 96 bytes in 4 blocks are definitely lost in loss record 6,432 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF48: Deck::Deck() (deck.cpp:41)
==22508==    by 0x20CCFB: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 96 bytes in 4 blocks are definitely lost in loss record 6,433 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF7C: Deck::Deck() (deck.cpp:42)
==22508==    by 0x20CCFB: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 96 bytes in 4 blocks are definitely lost in loss record 6,434 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF48: Deck::Deck() (deck.cpp:41)
==22508==    by 0x20D46F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 96 bytes in 4 blocks are definitely lost in loss record 6,435 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FF7C: Deck::Deck() (deck.cpp:42)
==22508==    by 0x20D46F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 102 bytes in 3 blocks are indirectly lost in loss record 6,437 of 7,517
==22508==    at 0x484DB80: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5CBB920: QArrayData::reallocateUnaligned(QArrayData*, void*, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5C98C7B: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5C92581: QString::append(QChar const*, long long) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5C9262F: QString::append(QString const&) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x21A350: QString::operator+=(QString const&) (qstring.h:710)
==22508==    by 0x21A6C3: operator+(char const*, QString const&) (qstring.h:1538)
==22508==    by 0x238D0F: Game::initPlayers() (game.cpp:95)
==22508==    by 0x238B5A: Game::setupBoard() (game.cpp:71)
==22508==    by 0x23894F: Game::Game(QObject*, GameParameters*) (game.cpp:27)
==22508==    by 0x1FC0B0: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:162)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508== 
==22508== 130 bytes in 1 blocks are indirectly lost in loss record 6,741 of 7,517
==22508==    at 0x484DB80: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5CBB920: QArrayData::reallocateUnaligned(QArrayData*, void*, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x23358A: QTypedArrayData<Card*>::reallocateUnaligned(QTypedArrayData<Card*>*, Card**, long long, QArrayData::AllocationOption) (qarraydata.h:117)
==22508==    by 0x232F8C: QtPrivate::QPodArrayOps<Card*>::reallocate(long long, QArrayData::AllocationOption) (qarraydataops.h:259)
==22508==    by 0x2327F2: QArrayDataPointer<Card*>::reallocateAndGrow(QArrayData::GrowthPosition, long long, QArrayDataPointer<Card*>*) (qarraydatapointer.h:208)
==22508==    by 0x232B1A: QArrayDataPointer<Card*>::detachAndGrow(QArrayData::GrowthPosition, long long, Card* const**, QArrayDataPointer<Card*>*) (qarraydatapointer.h:194)
==22508==    by 0x232208: void QtPrivate::QPodArrayOps<Card*>::emplace<Card*&>(long long, Card*&) (qarraydataops.h:174)
==22508==    by 0x231707: Card*& QList<Card*>::emplaceBack<Card*&>(Card*&) (qlist.h:856)
==22508==    by 0x230E12: QList<Card*>::append(Card*) (qlist.h:433)
==22508==    by 0x230A08: QList<Card*>::push_back(Card*) (qlist.h:651)
==22508==    by 0x236746: Player::obtainCard(Card*) (player.cpp:52)
==22508==    by 0x235C08: Human::drawCard(int const&) (human.cpp:45)
==22508== 
==22508== 152 bytes in 1 blocks are still reachable in loss record 6,797 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5D23C01: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5CA44ED: QThread::currentThread() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5B9019E: QCoreApplicationPrivate::QCoreApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x522B858: QGuiApplicationPrivate::QGuiApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x5237C2D: QGuiApplication::QGuiApplication(int&, char**, int) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x157332: main (main.cpp:7)
==22508== 
==22508== 170 (32 direct, 138 indirect) bytes in 1 blocks are definitely lost in loss record 6,856 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x2089BD: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:321)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 192 bytes in 8 blocks are indirectly lost in loss record 6,924 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FDE7: Deck::Deck() (deck.cpp:26)
==22508==    by 0x238ED3: Game::initDeck() (game.cpp:108)
==22508==    by 0x238B37: Game::setupBoard() (game.cpp:67)
==22508==    by 0x23894F: Game::Game(QObject*, GameParameters*) (game.cpp:27)
==22508==    by 0x1FC0B0: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:162)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508== 
==22508== 200 bytes in 1 blocks are still reachable in loss record 6,926 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5CA5047: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5D23C5D: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5CA44ED: QThread::currentThread() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x5B9019E: QCoreApplicationPrivate::QCoreApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x522B858: QGuiApplicationPrivate::QGuiApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x5237C2D: QGuiApplication::QGuiApplication(int&, char**, int) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==22508==    by 0x157332: main (main.cpp:7)
==22508== 
==22508== 250 (112 direct, 138 indirect) bytes in 1 blocks are definitely lost in loss record 6,996 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x206978: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:64)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 264 bytes in 11 blocks are indirectly lost in loss record 7,039 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FE25: Deck::Deck() (deck.cpp:27)
==22508==    by 0x238ED3: Game::initDeck() (game.cpp:108)
==22508==    by 0x238B37: Game::setupBoard() (game.cpp:67)
==22508==    by 0x23894F: Game::Game(QObject*, GameParameters*) (game.cpp:27)
==22508==    by 0x1FC0B0: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:162)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508== 
==22508== 288 bytes in 12 blocks are definitely lost in loss record 7,059 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEAB: Deck::Deck() (deck.cpp:33)
==22508==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 288 bytes in 12 blocks are definitely lost in loss record 7,060 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEE7: Deck::Deck() (deck.cpp:34)
==22508==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 288 bytes in 12 blocks are definitely lost in loss record 7,061 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEAB: Deck::Deck() (deck.cpp:33)
==22508==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 288 bytes in 12 blocks are definitely lost in loss record 7,062 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEE7: Deck::Deck() (deck.cpp:34)
==22508==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 288 bytes in 12 blocks are definitely lost in loss record 7,063 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEAB: Deck::Deck() (deck.cpp:33)
==22508==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 288 bytes in 12 blocks are definitely lost in loss record 7,064 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEE7: Deck::Deck() (deck.cpp:34)
==22508==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 288 bytes in 12 blocks are definitely lost in loss record 7,065 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEAB: Deck::Deck() (deck.cpp:33)
==22508==    by 0x20CCFB: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 288 bytes in 12 blocks are definitely lost in loss record 7,066 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEE7: Deck::Deck() (deck.cpp:34)
==22508==    by 0x20CCFB: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 288 bytes in 12 blocks are definitely lost in loss record 7,067 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEAB: Deck::Deck() (deck.cpp:33)
==22508==    by 0x20D46F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 288 bytes in 12 blocks are definitely lost in loss record 7,068 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FEE7: Deck::Deck() (deck.cpp:34)
==22508==    by 0x20D46F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 390 bytes in 3 blocks are indirectly lost in loss record 7,138 of 7,517
==22508==    at 0x484DB80: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5CBB920: QArrayData::reallocateUnaligned(QArrayData*, void*, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x23358A: QTypedArrayData<Card*>::reallocateUnaligned(QTypedArrayData<Card*>*, Card**, long long, QArrayData::AllocationOption) (qarraydata.h:117)
==22508==    by 0x232F8C: QtPrivate::QPodArrayOps<Card*>::reallocate(long long, QArrayData::AllocationOption) (qarraydataops.h:259)
==22508==    by 0x2327F2: QArrayDataPointer<Card*>::reallocateAndGrow(QArrayData::GrowthPosition, long long, QArrayDataPointer<Card*>*) (qarraydatapointer.h:208)
==22508==    by 0x232B1A: QArrayDataPointer<Card*>::detachAndGrow(QArrayData::GrowthPosition, long long, Card* const**, QArrayDataPointer<Card*>*) (qarraydatapointer.h:194)
==22508==    by 0x232208: void QtPrivate::QPodArrayOps<Card*>::emplace<Card*&>(long long, Card*&) (qarraydataops.h:174)
==22508==    by 0x231707: Card*& QList<Card*>::emplaceBack<Card*&>(Card*&) (qlist.h:856)
==22508==    by 0x230E12: QList<Card*>::append(Card*) (qlist.h:433)
==22508==    by 0x230A08: QList<Card*>::push_back(Card*) (qlist.h:651)
==22508==    by 0x236746: Player::obtainCard(Card*) (player.cpp:52)
==22508==    by 0x23B716: PC::drawCard(int const&) (pc.cpp:27)
==22508== 
==22508== 442 (112 direct, 330 indirect) bytes in 1 blocks are definitely lost in loss record 7,179 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x238A72: Game::setupParams(GameParameters*) (game.cpp:54)
==22508==    by 0x238943: Game::Game(QObject*, GameParameters*) (game.cpp:25)
==22508==    by 0x1FC0B0: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:162)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508== 
==22508== 864 bytes in 36 blocks are definitely lost in loss record 7,296 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FDE7: Deck::Deck() (deck.cpp:26)
==22508==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 864 bytes in 36 blocks are definitely lost in loss record 7,297 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FDE7: Deck::Deck() (deck.cpp:26)
==22508==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 864 bytes in 36 blocks are definitely lost in loss record 7,298 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FDE7: Deck::Deck() (deck.cpp:26)
==22508==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 864 bytes in 36 blocks are definitely lost in loss record 7,299 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FDE7: Deck::Deck() (deck.cpp:26)
==22508==    by 0x20CCFB: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 864 bytes in 36 blocks are definitely lost in loss record 7,300 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FDE7: Deck::Deck() (deck.cpp:26)
==22508==    by 0x20D46F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 936 bytes in 39 blocks are definitely lost in loss record 7,307 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FE25: Deck::Deck() (deck.cpp:27)
==22508==    by 0x20D46F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 960 bytes in 40 blocks are definitely lost in loss record 7,312 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FE25: Deck::Deck() (deck.cpp:27)
==22508==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 960 bytes in 40 blocks are definitely lost in loss record 7,313 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FE25: Deck::Deck() (deck.cpp:27)
==22508==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 960 bytes in 40 blocks are definitely lost in loss record 7,314 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FE25: Deck::Deck() (deck.cpp:27)
==22508==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 960 bytes in 40 blocks are definitely lost in loss record 7,315 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x22FE25: Deck::Deck() (deck.cpp:27)
==22508==    by 0x20CCFB: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508== 
==22508== 1,026 bytes in 1 blocks are indirectly lost in loss record 7,335 of 7,517
==22508==    at 0x484DB80: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x5CBB920: QArrayData::reallocateUnaligned(QArrayData*, void*, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==22508==    by 0x23358A: QTypedArrayData<Card*>::reallocateUnaligned(QTypedArrayData<Card*>*, Card**, long long, QArrayData::AllocationOption) (qarraydata.h:117)
==22508==    by 0x232F8C: QtPrivate::QPodArrayOps<Card*>::reallocate(long long, QArrayData::AllocationOption) (qarraydataops.h:259)
==22508==    by 0x2327F2: QArrayDataPointer<Card*>::reallocateAndGrow(QArrayData::GrowthPosition, long long, QArrayDataPointer<Card*>*) (qarraydatapointer.h:208)
==22508==    by 0x232B1A: QArrayDataPointer<Card*>::detachAndGrow(QArrayData::GrowthPosition, long long, Card* const**, QArrayDataPointer<Card*>*) (qarraydatapointer.h:194)
==22508==    by 0x232208: void QtPrivate::QPodArrayOps<Card*>::emplace<Card*&>(long long, Card*&) (qarraydataops.h:174)
==22508==    by 0x231707: Card*& QList<Card*>::emplaceBack<Card*&>(Card*&) (qlist.h:856)
==22508==    by 0x230E12: QList<Card*>::append(Card*) (qlist.h:433)
==22508==    by 0x230A08: QList<Card*>::push_back(Card*) (qlist.h:651)
==22508==    by 0x235E09: Pile::pushCard(Card*) (pile.cpp:17)
==22508==    by 0x202530: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:41)
==22508== 
==22508== 1,082 (32 direct, 1,050 indirect) bytes in 1 blocks are definitely lost in loss record 7,347 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x2024D6: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:36)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==22508==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==22508==    by 0x15735F: main (main.cpp:8)
==22508== 
==22508== 1,332 (336 direct, 996 indirect) bytes in 3 blocks are definitely lost in loss record 7,372 of 7,517
==22508==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==22508==    by 0x238D1D: Game::initPlayers() (game.cpp:95)
==22508==    by 0x238B5A: Game::setupBoard() (game.cpp:71)
==22508==    by 0x23894F: Game::Game(QObject*, GameParameters*) (game.cpp:27)
==22508==    by 0x1FC0B0: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:162)
==22508==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==22508==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==22508==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==22508==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==22508==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==22508==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==22508==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==22508== 
==22508== LEAK SUMMARY:
==22508==    definitely lost: 13,816 bytes in 555 blocks
==22508==    indirectly lost: 2,730 bytes in 49 blocks
==22508==      possibly lost: 0 bytes in 0 blocks
==22508==    still reachable: 690 bytes in 8 blocks
==22508==                       of which reachable via heuristic:
==22508==                         newarray           : 760 bytes in 5 blocks
==22508==         suppressed: 2,334,212 bytes in 22,869 blocks
==22508== 
==22508== For lists of detected and suppressed errors, rerun with: -s
==22508== ERROR SUMMARY: 50 errors from 50 contexts (suppressed: 55 from 55)
